MVC/view/CLIrefactor.py

```diff
@@ -40,41 +45,73 @@ def showHoneyComb(self):
+        Prompt class:
+            prompt: This libraries wayh to get input from the user
+            prompt(message,completer)
+                message: some type of text for the user to see
+                completer: this is built into the library defined by it, allows the user to press tab and autocomplete the commands we give it)
+                Due to this, this is why we set cmdautocomplete = wrdcmp(self.b4commands,ignore_case=True,match_middle=True) because it will allow tab completion to
+                happen with word completer.
+        '''
         while (True):
-            userInput = input("Please enter a guess or command, commands start with '!': ")
+            if self.controller.controllerGetPuzzleState() == 1:
+                 #once a game is started the user will have access to all of the commands, so uses this instead.
+                 cmdautocomplete = wrdcmp(self.commands,ignore_case=True,match_middle=True)
+            userInput = prompt("Please enter a guess or command, commands start with '!': ",completer=cmdautocomplete)
             checkInput = self.controller.checkInputCLI(userInput)
             while(checkInput == False):
-                userInput = input("Input can only contain [!, A-Z], please reenter: ")
+                userInput = prompt("Input can only contain [!, A-Z], please reenter: ",completer=cmdautocomplete)
                 checkInput = self.controller.checkInputCLI(userInput)
             while(len(userInput) < 4) or (len(userInput) > 15):
-                userInput = input("Input must be between 4 and 15 characters! Please reenter your input: ")
+                userInput = prompt("Input must be between 4 and 15 characters! Please reenter your input: ",completer=cmdautocomplete)
+
+
+
+            '''
+            In the scenario that a user types in multiple exclamation points, or when using tab completion theres still multiple
+            We can just check if the first spot in the string is an exclamation point and if the count of them is > 1, if so we can replace all !'s and just readd it at the end
+            '''
+            if (userInput[0] == '!') and userInput.count('!') > 1:
+                 # replace all !'s with a space, then add a '!' back into the front of it.
+                 rstring = userInput.replace("!","")
+                 fstring = '!' + rstring
+                 userInput = fstring
 
             if '!' not in userInput:
             # controller user guess function
                 if self.controller.controllerGetPuzzleState() != 1:
                     print('''
-To get started, you can type:
-    !newpuzzle: To generate a new puzzle. You can even provide your own pangram for puzzle creation!
-    !loadpuzzle: To load a saved puzzle from a file. You will need to enter the file name of the saved puzzle.
-    !help: To see the list of all the commands.
-    !exit: To exit the program.
-''')
+                    To get started, you can type:
```

Comment:
did you mean to reformat this? it's not consistent with the text above. please fix!